library/ssl_tls.c

```diff
@@ -2042,6 +2079,29 @@ static int ssl_tls13_session_load( mbedtls_ssl_session *session,
     memcpy( session->resumption_key, p, session->resumption_key_len );
     p += session->resumption_key_len;
 
+#if defined(MBEDTLS_SSL_SERVER_NAME_INDICATION) && defined(MBEDTLS_SSL_CLI_C)
+    if( session->endpoint == MBEDTLS_SSL_IS_CLIENT )
+    {
+        size_t hostname_len;
+        /* load host name */
+        if( end - p < 2 )
+            return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
+        hostname_len = MBEDTLS_GET_UINT16_BE( p, 0);
+        p += 2;
+
+        if( end - p < ( long int )hostname_len )
```

Comment:
```suggestion
        if( end - p < hostname_len )
```
Why would we need this cast ?

Reply:
the type of (end - p) is long int, but the hostmame_len be defined as size_t.